projects/HW102/src/main.c

```diff
@@ -0,0 +1,37 @@
+#include <stdint.h>
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+#define SECOND 1000
+#define HALF_SECOND 500
+
+typedef struct Counters {
+  uint16_t counter_a;
+  uint16_t counter_b;
+} Counters;
+
+void increment_a(SoftTimerId timer_id, void *context) {
+  Counters *storage = context;
+  storage->counter_a++;
+  LOG_DEBUG("%d\n", storage->counter_a);
+  soft_timer_start_millis(HALF_SECOND, increment_a, &storage, NULL);
+}
+
+void increment_b(SoftTimerId timer_id, void *context) {
+  Counters *storage = context;
+  storage->counter_b++;
+  LOG_DEBUG("%d\n", storage->counter_b);
+  soft_timer_start_millis(SECOND, increment_b, &storage, NULL);
+}
+
+int main() {
```

Comment:
You'll need to initialize interrupts and soft timers here. You can also test your FW102 with `make run PLATFORM=x86 PROJECT=HW102`.